packages/ckeditor5-dev-build-tools/src/plugins/cssStyles.ts

```diff
@@ -0,0 +1,305 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { createFilter } from '@rollup/pluginutils';
+import { parse, type Rule, type Declaration, type Stylesheet } from 'css';
+import type { Plugin, GetModuleInfo, OutputBundle, OutputChunk, NormalizedOutputOptions } from 'rollup';
+
+/**
+ * Filter files only with `css` extension.
+ */
+const filter = createFilter( [ '**/*.css' ] );
+
+export function cssStyles(): Plugin {
+	const styles: Record<string, string> = {};
+
+	return {
+		name: 'ck5-styles',
+		transform( code, id ) {
+			if ( !filter( id ) ) {
+				return;
+			}
+
+			styles[ id ] = code;
+
+			return '';
+		},
+		// ////
+		// `moduleParsed` is just for debugging.
+		// ////
+		moduleParsed( moduleInfo ) {
+			console.log( `Module ${ moduleInfo.id } has been parsed.` );
+		},
+		async generateBundle( output: NormalizedOutputOptions, bundle: OutputBundle ) {
+			const banner = await getBanner( output, bundle );
+			const ids: Set<string> = new Set();
+
+			// Determine import order of files
+			for ( const file in bundle ) {
+				const outputChunk = bundle[ file ]! as OutputChunk;
+				const root = outputChunk.facadeModuleId!;
+				const modules = getCSSModules( root, this.getModuleInfo );
+
+				modules.forEach( id => ids.add( id as string ) );
+			}
+
+			const cssStylesheet = combineStylesheetsIntoOne( ids, styles );
```

Comment:
This is just an idea, but we might be able to use `MagicString.Bundle' to concatenate multiple files. This would not only simplify our implementation, but we would also be able to get generate proper source maps, which are invaluable when debugging the code.

I also used this library in other plugin (still WIP), so we'll use is anyway.

https://github.com/rich-harris/magic-string?tab=readme-ov-file#bundling